app/controllers/site_controller.rb

```diff
@@ -102,7 +103,9 @@ def copyright
     @locale = params[:copyright_locale] || I18n.locale
   end
 
-  def welcome; end
+  def welcome
+    require_user
```

Comment:
Could you explain this bit for me? My thoughts were that using CanCanCan would be to remove uses of our home-grown require_user/admin/moderator from the code. My approach would be to change the authorisation rules to have `can [:welcome], :site` in the "if user" section, rather than putting it in the 'all users' section and then adding a restriction here. Have I overlooked something?